core/engine.go

```diff
@@ -299,7 +311,7 @@ func (e *Engine) processMetrics(globalCtx context.Context, processMetricsAfterRu
 			if !e.runtimeOptions.NoThresholds.Bool {
 				// Ensure the ingester flushes any buffered metrics
 				_ = e.ingester.Stop()
-				thresholdsTainted, _ := e.MetricsEngine.EvaluateThresholds()
+				thresholdsTainted, _ := e.MetricsEngine.EvaluateThresholds(true)
```

Comment:
I think you continued my mistake with the boolean logic :rofl:. 

We don't should ignore empty sinks everywhere except here. Also my `IsEmpty` code is all wrong ... so you basically need to reverse to logic (or check for `== 0` instead of `>0` ).

 ```suggestion
				thresholdsTainted, _ := e.MetricsEngine.EvaluateThresholds(false)
```